cuda_phase-retrieval_class/test/test_cupy.hpp

```diff
@@ -147,4 +166,28 @@ void test_cupy_from_c_memory()
     assert(py::module::import("cupy").attr("get_default_memory_pool")().attr("used_bytes")().cast<size_t>() > 0 );
 
     //cp removed automatically after the next closing bracket
+}
+
+template<typename TInputData, typename TOutputData>
+TOutputData * convertToCUFFT(TInputData * ptr)
+{
+  throw std::logic_error("Not implemented");
+}
+
+template<>
+cufftDoubleComplex *convertToCUFFT(complex<double> * ptr)
+{  
+    return reinterpret_cast<cufftDoubleComplex *>(ptr);
+}
+
+template <typename T>
+Custom_Cupy_Ref<T> getCustomCupyRef(py::object obj)
+{
+    if(py::module::import("builtins").attr("str")(obj.attr("__class__")).cast<string>() != "<class 'cupy.core.core.ndarray'>")
+    {
+        throw std::exception("Exception : Python object must be a cupy array");
```

Comment:
Replace it with a `runtime_error` or something similar. Normally `std::exception` has no constructor with a string argument.